### DISCUSSION TITLE
How can I fix the 'wrong row length' error when adding a row to a CIF file with the gemmi library in Python?

### DISCUSSION THREAD

**Discussion by lobnaOmar20** [project-gemmi/gemmi#266] [Q&A] [answered]
why everytime I try to add a row to my cif file using this:

```
loop = block.find_loop('_atom_type_symbol').get_loop()
loop.add_row(['Au'], pos=0)
loop.add_row(['Zr'])  # appends
list(block.find_loop('_atom_type.symbol'))
['Au', 'C', 'CL', 'N', 'O', 'P', 'S', 'Zr']
```
which is exactly the same as the one in the documentation 
I get
(RuntimeError: add_row(): wrong row length.)

I tried different positions in here
`loop.add_row(['Au'], pos=0) 
`
with no better result, I always get the same error.

I used gemmi for a lot of things for my cif files. It is awesome. 

I just want to add sodium atoms to my cif file.

**Comment by wojdyr** [comment]
In mmCIF files _atom_type category usually has one column with symbol.
In small molecule CIF files it often has multiple columns, such as
```
loop_
 _atom_type_symbol
 _atom_type_description
 _atom_type_scat_dispersion_real
 _atom_type_scat_dispersion_imag
 _atom_type_scat_source
```
The list argument to `add_row()` needs to have as many values as there are columns in the loop.

**Reply by lobnaOmar20** [reply]
ahhhaaa ... let me try this and I will keep you updated. 
thank you very much for your quick reply. 
gemmi library is awesome. I love it !!! It helps me so much

**Reply by lobnaOmar20** [reply]
you mean cif.quotelist() ?? I tried with that as well but no better result.

**Reply by lobnaOmar20** [reply]
```
loop = block.find_loop_item('_atom_site_label').loop
loop.tags 
```
['_atom_site_label',
 '_atom_site_occupancy',
 '_atom_site_fract_x',
 '_atom_site_fract_y',
 '_atom_site_fract_z',
 '_atom_site_adp_type',
 '_atom_site_U_iso_or_equiv',
 '_atom_site_type_symbol']

should I do this??
`loop.add_row(['Na1'],['1.0'],[0.141800],[0.000000],[0.667300],['Uiso'],[0.050000],[sodium]) `

**Reply by lobnaOmar20** [reply]
tried this as well, but I always get an error 

add_row(): incompatible function arguments. The following argument types are supported:
    1. (self: gemmi.cif.Loop, new_values: List[str], pos: int = -1) -> None

**Reply by wojdyr** [reply, marked as answer]
one list of strings

    loop.add_row(['Na1' , '1.0', '0.141800', '0.000000', '0.667300', 'Uiso', '0.050000', 'sodium'])

**Reply by lobnaOmar20** [reply]
thank you soooo much it finally worked.
